### ISSUE TITLE
bad_alloc error

### ISSUE THREAD

**Issue by baekg** [cvxgrp/cvxpylayers#84]
Hi, I am getting bad_alloc error on executing following piece of code. If I run with m=300,p=400 it runs ok. Is there any solution or any hack to avoid running into memory issues like this?

```
import cvxpy as cp
import torch 
from cvxpylayers.torch import CvxpyLayer
n, m,p = 64, 3000,4000
x = cp.Variable((p))
w = cp.Parameter((p))
b = cp.Parameter((m))
A = cp.Parameter((m,p))
constraints = [A@x==b]
objective = cp.Minimize(w@x)
problem = cp.Problem(objective, constraints)
assert problem.is_dpp()
cvxpylayer = CvxpyLayer(problem, parameters=[A,w,b], variables=[x])

```

**Comment by jinga-lala** [comment]
Did any find any solution for this problem?

**Comment by akshayka** [comment]
CVXPY unfortunately cannot canonicalize problems with many parameters; memory becomes a bottleneck. This is an issue that the CVXPY developers are aware of.

The only thing I would suggest is to make A sparse instead of dense if possible. See this thread for how to create sparse parameters: https://github.com/cvxgrp/cvxpy/issues/1159

**Comment by sbarratt** [comment]
As @akshayka said, you either need a beefier machine with more memory, to decrease your problem size, or make your parameters sparse. There were some improvements made to cvxpy compilation and memory usage, so it might be worth upgrading cvxpy and giving it a try again:
```
pip install --upgrade cvxpy
```

I'm closing this issue.

**Comment by lemon234071** [comment]
Same problem and i have 3T memory 
  
    m = 2000
    n = 2000
    np.random.seed(1)
    A = cp.Parameter((m, n))
    b = cp.Parameter(m)
    A.value = np.random.randn(m, n)
    b.value = np.random.randn(m)
    # A = np.random.randn(m, n)
    # b = np.random.randn(m)
    
    # Construct the problem.
    x = cp.Variable(n)
    objective = cp.Minimize(cp.sum_squares(A @ x - b))
    constraints = [0 <= x, x <= 1]
    prob = cp.Problem(objective, constraints)
    
    # The optimal objective value is returned by `prob.solve()`.
    result = prob.solve(verbose=True)
    # The optimal value for x is stored in `x.value`.
    print(x.value)
    # The optimal Lagrange multiplier for a constraint is stored in
    # `constraint.dual_value`.
    print(constraints[0].dual_value)
    print(time.time() - s_t)
